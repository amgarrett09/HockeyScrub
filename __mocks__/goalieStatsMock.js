const stats = [
  {
    season: '20182019',
    stat: {
      timeOnIce: '1658:22',
      ot: 1,
      shutouts: 0,
      ties: 0,
      wins: 16,
      losses: 12,
      saves: 871,
      powerPlaySaves: 83,
      shortHandedSaves: 30,
      evenSaves: 758,
      shortHandedShots: 31,
      evenShots: 825,
      powerPlayShots: 92,
      savePercentage: 0.919,
      goalAgainstAverage: 2.79,
      games: 30,
      gamesStarted: 29,
      shotsAgainst: 948,
      goalsAgainst: 77,
      powerPlaySavePercentage: 90.21739130434783,
      shortHandedSavePercentage: 96.7741935483871,
      evenStrengthSavePercentage: 91.87878787878788,
    },
    team: {
      id: 4,
      name: 'Philadelphia Flyers',
      link: '/api/v1/teams/4',
    },
    league: {
      id: 133,
      name: 'National Hockey League',
      link: '/api/v1/league/133',
    },
    sequenceNumber: 1,
  },
];

const career = {
  timeOnIce: '1658:22',
  ot: 1,
  shutouts: 0,
  wins: 16,
  losses: 12,
  saves: 871,
  powerPlaySaves: 83,
  shortHandedSaves: 30,
  evenSaves: 758,
  shortHandedShots: 31,
  evenShots: 825,
  powerPlayShots: 92,
  savePercentage: 0.918776,
  goalAgainstAverage: 2.78587,
  games: 30,
  gamesStarted: 29,
  shotsAgainst: 948,
  goalsAgainst: 77,
  timeOnIcePerGame: '55:16',
  powerPlaySavePercentage: 90.21739130434783,
  shortHandedSavePercentage: 96.7741935483871,
  evenStrengthSavePercentage: 91.87878787878788,
};

module.exports = {
  stats,
  career,
};
