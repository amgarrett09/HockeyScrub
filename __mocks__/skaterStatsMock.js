const stats = [
  {
    season: '20162017',
    stat: {
      timeOnIce: '63:45',
      assists: 0,
      goals: 0,
      pim: 0,
      shots: 3,
      games: 7,
      hits: 12,
      powerPlayGoals: 0,
      powerPlayPoints: 0,
      powerPlayTimeOnIce: '00:22',
      evenTimeOnIce: '63:23',
      penaltyMinutes: '0',
      faceOffPct: 50.0,
      shotPct: 0.0,
      gameWinningGoals: 0,
      overTimeGoals: 0,
      shortHandedGoals: 0,
      shortHandedPoints: 0,
      shortHandedTimeOnIce: '00:00',
      blocked: 0,
      plusMinus: -1,
      points: 0,
      shifts: 90,
    },
    team: {
      id: 17,
      name: 'Detroit Red Wings',
      link: '/api/v1/teams/17',
    },
    league: {
      id: 133,
      name: 'National Hockey League',
      link: '/api/v1/league/133',
    },
    sequenceNumber: 1,
  },
  {
    season: '20162017',
    stat: {
      assists: 25,
      goals: 12,
      pim: 37,
      games: 48,
      penaltyMinutes: '37',
      plusMinus: 15,
      points: 37,
    },
    team: {
      name: 'Grand Rapids Griffins',
      link: '/api/v1/teams/null',
    },
    league: {
      name: 'AHL',
      link: '/api/v1/league/null',
    },
    sequenceNumber: 978,
  },
  {
    season: '20172018',
    stat: {
      timeOnIce: '691:12',
      assists: 17,
      goals: 7,
      pim: 39,
      shots: 83,
      games: 48,
      hits: 53,
      powerPlayGoals: 1,
      powerPlayPoints: 2,
      powerPlayTimeOnIce: '68:25',
      evenTimeOnIce: '617:34',
      penaltyMinutes: '39',
      faceOffPct: 12.5,
      shotPct: 8.4,
      gameWinningGoals: 1,
      overTimeGoals: 0,
      shortHandedGoals: 0,
      shortHandedPoints: 0,
      shortHandedTimeOnIce: '05:13',
      blocked: 38,
      plusMinus: -7,
      points: 24,
      shifts: 901,
    },
    team: {
      id: 17,
      name: 'Detroit Red Wings',
      link: '/api/v1/teams/17',
    },
    league: {
      id: 133,
      name: 'National Hockey League',
      link: '/api/v1/league/133',
    },
    sequenceNumber: 1,
  },
  {
    season: '20172018',
    stat: {
      assists: 7,
      goals: 7,
      pim: 34,
      games: 16,
      penaltyMinutes: '34',
      plusMinus: -5,
      points: 14,
    },
    team: {
      name: 'Grand Rapids Griffins',
      link: '/api/v1/teams/null',
    },
    league: {
      name: 'AHL',
      link: '/api/v1/league/null',
    },
    sequenceNumber: 978,
  },
  {
    season: '20182019',
    stat: {
      timeOnIce: '1184:04',
      assists: 26,
      goals: 21,
      pim: 36,
      shots: 130,
      games: 72,
      hits: 74,
      powerPlayGoals: 3,
      powerPlayPoints: 6,
      powerPlayTimeOnIce: '121:37',
      evenTimeOnIce: '1061:16',
      penaltyMinutes: '36',
      faceOffPct: 33.33,
      shotPct: 16.2,
      gameWinningGoals: 4,
      overTimeGoals: 1,
      shortHandedGoals: 0,
      shortHandedPoints: 0,
      shortHandedTimeOnIce: '01:11',
      blocked: 38,
      plusMinus: 12,
      points: 47,
      shifts: 1525,
    },
    team: {
      id: 17,
      name: 'Detroit Red Wings',
      link: '/api/v1/teams/17',
    },
    league: {
      id: 133,
      name: 'National Hockey League',
      link: '/api/v1/league/133',
    },
    sequenceNumber: 1,
  },
];

const career = {
  timeOnIce: '1939:01',
  assists: 43,
  goals: 28,
  pim: 75,
  shots: 216,
  games: 127,
  hits: 139,
  powerPlayGoals: 4,
  powerPlayPoints: 8,
  powerPlayTimeOnIce: '190:24',
  evenTimeOnIce: '1742:13',
  penaltyMinutes: '75',
  faceOffPct: 30.0,
  shotPct: 13.0,
  gameWinningGoals: 5,
  overTimeGoals: 1,
  shortHandedGoals: 0,
  shortHandedPoints: 0,
  shortHandedTimeOnIce: '06:24',
  blocked: 76,
  plusMinus: 4,
  points: 71,
  shifts: 2516,
  timeOnIcePerGame: '15:16',
  evenTimeOnIcePerGame: '13:43',
  shortHandedTimeOnIcePerGame: '00:03',
  powerPlayTimeOnIcePerGame: '01:29',
};

const lastFive = [
  {
    season: '20182019',
    stat: {
      timeOnIce: '20:25',
      assists: 0,
      goals: 0,
      pim: 4,
      shots: 4,
      games: 1,
      hits: 1,
      powerPlayGoals: 0,
      powerPlayPoints: 0,
      powerPlayTimeOnIce: '02:40',
      evenTimeOnIce: '17:45',
      penaltyMinutes: '4',
      shotPct: 0.0,
      gameWinningGoals: 0,
      overTimeGoals: 0,
      shortHandedGoals: 0,
      shortHandedPoints: 0,
      shortHandedTimeOnIce: '00:00',
      blocked: 0,
      plusMinus: -1,
      points: 0,
      shifts: 24,
    },
    team: {
      id: 17,
      name: 'Detroit Red Wings',
      link: '/api/v1/teams/17',
    },
    opponent: {
      id: 5,
      name: 'Pittsburgh Penguins',
      link: '/api/v1/teams/5',
    },
    date: '2019-04-04',
    isHome: false,
    isWin: false,
    isOT: false,
    game: {
      gamePk: 2018021245,
      link: '/api/v1/game/2018021245/feed/live',
      content: {
        link: '/api/v1/game/2018021245/content',
      },
    },
  }, {
    season: '20182019',
    stat: {
      timeOnIce: '19:26',
      assists: 1,
      goals: 2,
      pim: 0,
      shots: 5,
      games: 1,
      hits: 0,
      powerPlayGoals: 1,
      powerPlayPoints: 1,
      powerPlayTimeOnIce: '00:22',
      evenTimeOnIce: '18:48',
      penaltyMinutes: '0',
      shotPct: 40.0,
      gameWinningGoals: 1,
      overTimeGoals: 0,
      shortHandedGoals: 0,
      shortHandedPoints: 0,
      shortHandedTimeOnIce: '00:16',
      blocked: 0,
      plusMinus: 2,
      points: 3,
      shifts: 27,
    },
    team: {
      id: 17,
      name: 'Detroit Red Wings',
      link: '/api/v1/teams/17',
    },
    opponent: {
      id: 5,
      name: 'Pittsburgh Penguins',
      link: '/api/v1/teams/5',
    },
    date: '2019-04-02',
    isHome: true,
    isWin: true,
    isOT: false,
    game: {
      gamePk: 2018021233,
      link: '/api/v1/game/2018021233/feed/live',
      content: {
        link: '/api/v1/game/2018021233/content',
      },
    },
  }, {
    season: '20182019',
    stat: {
      timeOnIce: '16:18',
      assists: 3,
      goals: 0,
      pim: 0,
      shots: 1,
      games: 1,
      hits: 0,
      powerPlayGoals: 0,
      powerPlayPoints: 2,
      powerPlayTimeOnIce: '04:20',
      evenTimeOnIce: '11:51',
      penaltyMinutes: '0',
      shotPct: 0.0,
      gameWinningGoals: 0,
      overTimeGoals: 0,
      shortHandedGoals: 0,
      shortHandedPoints: 0,
      shortHandedTimeOnIce: '00:07',
      blocked: 1,
      plusMinus: 1,
      points: 3,
      shifts: 22,
    },
    team: {
      id: 17,
      name: 'Detroit Red Wings',
      link: '/api/v1/teams/17',
    },
    opponent: {
      id: 6,
      name: 'Boston Bruins',
      link: '/api/v1/teams/6',
    },
    date: '2019-03-31',
    isHome: true,
    isWin: true,
    isOT: false,
    game: {
      gamePk: 2018021219,
      link: '/api/v1/game/2018021219/feed/live',
      content: {
        link: '/api/v1/game/2018021219/content',
      },
    },
  }, {
    season: '20182019',
    stat: {
      timeOnIce: '16:19',
      assists: 2,
      goals: 1,
      pim: 0,
      shots: 4,
      games: 1,
      hits: 2,
      powerPlayGoals: 0,
      powerPlayPoints: 0,
      powerPlayTimeOnIce: '01:00',
      evenTimeOnIce: '15:19',
      penaltyMinutes: '0',
      shotPct: 25.0,
      gameWinningGoals: 0,
      overTimeGoals: 0,
      shortHandedGoals: 0,
      shortHandedPoints: 0,
      shortHandedTimeOnIce: '00:00',
      blocked: 0,
      plusMinus: 3,
      points: 3,
      shifts: 20,
    },
    team: {
      id: 17,
      name: 'Detroit Red Wings',
      link: '/api/v1/teams/17',
    },
    opponent: {
      id: 1,
      name: 'New Jersey Devils',
      link: '/api/v1/teams/1',
    },
    date: '2019-03-29',
    isHome: true,
    isWin: true,
    isOT: false,
    game: {
      gamePk: 2018021199,
      link: '/api/v1/game/2018021199/feed/live',
      content: {
        link: '/api/v1/game/2018021199/content',
      },
    },
  }, {
    season: '20182019',
    stat: {
      timeOnIce: '16:58',
      assists: 1,
      goals: 2,
      pim: 0,
      shots: 4,
      games: 1,
      hits: 0,
      powerPlayGoals: 1,
      powerPlayPoints: 1,
      powerPlayTimeOnIce: '02:46',
      evenTimeOnIce: '14:12',
      penaltyMinutes: '0',
      shotPct: 50.0,
      gameWinningGoals: 1,
      overTimeGoals: 1,
      shortHandedGoals: 0,
      shortHandedPoints: 0,
      shortHandedTimeOnIce: '00:00',
      blocked: 0,
      plusMinus: 1,
      points: 3,
      shifts: 21,
    },
    team: {
      id: 17,
      name: 'Detroit Red Wings',
      link: '/api/v1/teams/17',
    },
    opponent: {
      id: 7,
      name: 'Buffalo Sabres',
      link: '/api/v1/teams/7',
    },
    date: '2019-03-28',
    isHome: false,
    isWin: true,
    isOT: false,
    game: {
      gamePk: 2018021189,
      link: '/api/v1/game/2018021189/feed/live',
      content: {
        link: '/api/v1/game/2018021189/content',
      },
    },
  },
];

module.exports = {
  stats,
  career,
  lastFive,
};
