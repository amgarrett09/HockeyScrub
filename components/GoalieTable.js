import React, { useState } from 'react';
import PropTypes from 'prop-types';
import convertSeason from '../lib/convert-season';
import teamAbbreviations from '../constants/constants';

const sortingFunctions = {
  Season(a, b) {
    return a.season - b.season;
  },
  GP(a, b) {
    return b.stat.games - a.stat.games;
  },
  GS(a, b) {
    return b.stat.gamesStarted - a.stat.gamesStarted;
  },
  W(a, b) {
    return b.stat.wins - a.stat.wins;
  },
  L(a, b) {
    return b.stat.losses - a.stat.losses;
  },
  T(a, b) {
    return b.stat.ties - a.stat.ties;
  },
  OTL(a, b) {
    return b.stat.ot - a.stat.ot;
  },
  SA(a, b) {
    return b.stat.shotsAgainst - a.stat.shotsAgainst;
  },
  GA(a, b) {
    return b.stat.goalsAgainst - a.stat.goalsAgainst;
  },
  GAA(a, b) {
    return b.stat.goalAgainstAverage - a.stat.goalAgainstAverage;
  },
  S(a, b) {
    return b.stat.saves - a.stat.saves;
  },
  // eslint-disable-next-line func-names
  'S%': function (a, b) {
    return b.stat.savePercentage - a.stat.savePercentage;
  },
  SO(a, b) {
    return b.stat.shutouts - a.stat.shutouts;
  },
  MIN(a, b) {
    return b.stat.timeOnIce - a.stat.timeOnIce;
  },
};

const GoalieTable = ({ stats, career }) => {
  const [sortedStats, setSortedStats] = useState(stats);
  const [currentlySorted, setCurrentlySorted] = useState('');

  // Sort the table by a particular stat when you click on the column's title.
  const sortBy = (e) => {
    e.preventDefault();

    const colTitle = e.target.innerText;

    if (currentlySorted === colTitle) {
      const newStats = sortedStats.slice(0, sortedStats.length).reverse();

      setSortedStats(newStats);
    } else {
      const newStats = sortedStats
        .slice(0, sortedStats.length)
        .sort(sortingFunctions[colTitle]);

      setSortedStats(newStats);
      setCurrentlySorted(colTitle);
    }
  };

  return (
    <React.Fragment>
      <table className="table is-narrow is-hoverable is-fullwidth">
        <thead>
          <tr>
            <th>
              <button type="button" className="sort-button" onClick={sortBy}>
                Season
              </button>
            </th>
            <th>Team</th>
            <th>
              <button type="button" className="sort-button" onClick={sortBy}>
                <abbr title="Games Played">GP</abbr>
              </button>
            </th>
            <th>
              <button type="button" className="sort-button" onClick={sortBy}>
                <abbr title="Games Played">GS</abbr>
              </button>
            </th>
            <th>
              <button type="button" className="sort-button" onClick={sortBy}>
                <abbr title="Games Played">W</abbr>
              </button>
            </th>
            <th>
              <button type="button" className="sort-button" onClick={sortBy}>
                <abbr title="Games Played">L</abbr>
              </button>
            </th>
            <th>
              <button type="button" className="sort-button" onClick={sortBy}>
                <abbr title="Games Played">T</abbr>
              </button>
            </th>
            <th>
              <button type="button" className="sort-button" onClick={sortBy}>
                <abbr title="Games Played">OTL</abbr>
              </button>
            </th>
            <th>
              <button type="button" className="sort-button" onClick={sortBy}>
                <abbr title="Games Played">SA</abbr>
              </button>
            </th>
            <th>
              <button type="button" className="sort-button" onClick={sortBy}>
                <abbr title="Games Played">GA</abbr>
              </button>
            </th>
            <th>
              <button type="button" className="sort-button" onClick={sortBy}>
                <abbr title="Games Played">GAA</abbr>
              </button>
            </th>
            <th>
              <button type="button" className="sort-button" onClick={sortBy}>
                <abbr title="Games Played">S</abbr>
              </button>
            </th>
            <th>
              <button type="button" className="sort-button" onClick={sortBy}>
                <abbr title="Games Played">S%</abbr>
              </button>
            </th>
            <th>
              <button type="button" className="sort-button" onClick={sortBy}>
                <abbr title="Games Played">SO</abbr>
              </button>
            </th>
            <th>
              <button type="button" className="sort-button" onClick={sortBy}>
                <abbr title="Games Played">MIN</abbr>
              </button>
            </th>
          </tr>
        </thead>

        <tbody>
          {sortedStats.map((obj, i) => (
            <tr key={obj.season + obj.team.name + i.toString()}>
              <td
                className={currentlySorted === 'Season' ? 'stat-cell--sorted' : ''}
              >
                {convertSeason(obj.season)}
              </td>
              <td>{teamAbbreviations[obj.team.name] || obj.team.name}</td>
              <td
                className={currentlySorted === 'GP' ? 'stat-cell--sorted' : ''}
              >
                {obj.stat.games}
              </td>
              <td
                className={currentlySorted === 'GS' ? 'stat-cell--sorted' : ''}
              >
                {obj.stat.gamesStarted}
              </td>
              <td
                className={currentlySorted === 'W' ? 'stat-cell--sorted' : ''}
              >
                {obj.stat.wins}
              </td>
              <td
                className={currentlySorted === 'L' ? 'stat-cell--sorted' : ''}
              >
                {obj.stat.losses}
              </td>
              <td
                className={currentlySorted === 'T' ? 'stat-cell--sorted' : ''}
              >
                {obj.stat.ties}
              </td>
              <td
                className={currentlySorted === 'OTL' ? 'stat-cell--sorted' : ''}
              >
                {obj.stat.ot}
              </td>
              <td
                className={currentlySorted === 'SA' ? 'stat-cell--sorted' : ''}
              >
                {obj.stat.shotsAgainst}
              </td>
              <td
                className={currentlySorted === 'GA' ? 'stat-cell--sorted' : ''}
              >
                {obj.stat.goalsAgainst}
              </td>
              <td
                className={currentlySorted === 'GAA' ? 'stat-cell--sorted' : ''}
              >
                {obj.stat.goalAgainstAverage}
              </td>
              <td
                className={currentlySorted === 'S' ? 'stat-cell--sorted' : ''}
              >
                {obj.stat.saves}
              </td>
              <td
                className={currentlySorted === 'S%' ? 'stat-cell--sorted' : ''}
              >
                {obj.stat.savePercentage}
              </td>
              <td
                className={currentlySorted === 'SO' ? 'stat-cell--sorted' : ''}
              >
                {obj.stat.shutouts}
              </td>
              <td
                className={currentlySorted === 'MIN' ? 'stat-cell--sorted' : ''}
              >
                {obj.stat.timeOnIce}
              </td>
            </tr>
          ))}
          <tr>
            <td>
              <strong>Career</strong>
            </td>
            <td />
            <td>
              <strong>{career.games}</strong>
            </td>
            <td>
              <strong>{career.gamesStarted}</strong>
            </td>
            <td>
              <strong>{career.wins}</strong>
            </td>
            <td>
              <strong>{career.losses}</strong>
            </td>
            <td>
              <strong>{career.ties}</strong>
            </td>
            <td>
              <strong>{career.ot}</strong>
            </td>
            <td>
              <strong>{career.shotsAgainst}</strong>
            </td>
            <td>
              <strong>{career.goalsAgainst}</strong>
            </td>
            <td>
              <strong>{career.goalAgainstAverage}</strong>
            </td>
            <td>
              <strong>{career.saves}</strong>
            </td>
            <td>
              <strong>{career.savePercentage}</strong>
            </td>
            <td>
              <strong>{career.shutouts}</strong>
            </td>
            <td>
              <strong>{career.timeOnIce}</strong>
            </td>
          </tr>
        </tbody>
      </table>

      <style jsx>
        {`
          .sort-button {
            padding: 0;
            margin: 0;
            border: none;
            font-weight: bold;
            background: transparent;
          }
          .sort-button:hover {
            cursor: pointer;
          }

          .stat-cell--sorted {
            background-color: yellow;
          }
        `}
      </style>
    </React.Fragment>
  );
};

GoalieTable.propTypes = {
  stats: PropTypes.arrayOf(
    PropTypes.shape({
      season: PropTypes.string,
    }),
  ),
  career: PropTypes.shape({
    stat: PropTypes.shape({
      timeOnIce: PropTypes.string,
    }),
  }),
};

GoalieTable.defaultProps = {
  stats: [],
  career: {},
};

export default GoalieTable;
