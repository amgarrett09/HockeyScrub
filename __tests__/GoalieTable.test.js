/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';

import GoalieTable from '../components/GoalieTable';
import { stats, career } from '../__mocks__/goalieStatsMock';

describe('GoalieTable', () => {
  it('should match snapshot', () => {
    const component = renderer.create(
      <GoalieTable stats={stats} career={career} />,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
