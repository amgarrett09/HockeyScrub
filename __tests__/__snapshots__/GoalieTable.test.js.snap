// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoalieTable should match snapshot 1`] = `[Function]`;
